kratos/python_scripts/auxiliar_methods_adaptative_remeshing.py

```diff
@@ -0,0 +1,161 @@
+                ReInitializeSolver(analysis)
+            if non_linear_iteration == 1 or root_model_part.Is(KratosMultiphysics.MODIFIED):
+                analysis.InitializeSolutionStep()
+                solver.Predict()
+                root_model_part.Set(KratosMultiphysics.MODIFIED, False)
+                analysis.is_printing_rank = False
+            computing_model_part.ProcessInfo.SetValue(KratosMultiphysics.NL_ITERATION_NUMBER, non_linear_iteration)
+            is_converged = convergence_criteria.PreCriteria(computing_model_part, builder_and_solver.GetDofSet(), mechanical_solution_strategy.GetSystemMatrix(), mechanical_solution_strategy.GetSolutionVector(), mechanical_solution_strategy.GetSystemVector())
+            solver.SolveSolutionStep()
+            is_converged = convergence_criteria.PostCriteria(computing_model_part, builder_and_solver.GetDofSet(), mechanical_solution_strategy.GetSystemMatrix(), mechanical_solution_strategy.GetSolutionVector(), mechanical_solution_strategy.GetSystemVector())
+            if with_contact:
+                analysis._transfer_slave_to_master()
+            analysis.FinalizeSolutionStep()
+            if is_converged:
+                analysis.is_printing_rank = True
+                KratosMultiphysics.Logger.PrintInfo(analysis._GetSimulationName(), "Adaptative strategy converged in ", non_linear_iteration, "iterations" )
+                break
+            elif non_linear_iteration == analysis.non_linear_iterations:
+                analysis.is_printing_rank = True
+                KratosMultiphysics.Logger.PrintInfo(analysis._GetSimulationName(), "Adaptative strategy not converged after ", non_linear_iteration, "iterations" )
+                break
+            else:
+                if with_contact:
+                    # Before remesh we set the flag INTERFACE to the conditions (we need edges to preserve submodelparts)
+                    KratosMultiphysics.VariableUtils().SetFlag(KratosMultiphysics.INTERFACE, True, computing_model_part.GetSubModelPart("Contact").Conditions)
+
+                    # We remove the contact model part to avoid problems (it will  be recomputed later)
+                    contact_model_part = computing_model_part.GetSubModelPart("Contact")
+                    for model_part in contact_model_part.SubModelParts:
+                        contact_model_part.RemoveSubModelPart(model_part.Name)
+                    computing_model_part.RemoveSubModelPart("ComputingContact")
+
+                metric_process.Execute()
+                remeshing_process.Execute()
+
+                if with_contact:
+                    # We remove the contact model part to avoid problems (it will  be recomputed later)
+                    computing_model_part.RemoveSubModelPart("Contact")
+                root_model_part.Set(KratosMultiphysics.MODIFIED, True)
+                non_linear_iteration += 1
+        analysis.OutputSolutionStep()
+
+def ReInitializeSolver(analysis):
```

Comment:
Such a function can be useful in the base solver.

Reply:
That would take long to be accepted